doc/api/packages.md

```diff
@@ -819,10 +743,219 @@ conditional exports for consumers could be to add an export, e.g.
+    pr-url: https://github.com/nodejs/node/pull/31002
+    description: Remove the `--experimental-resolve-self` option.
+-->
+
+```json
+{
+  "name": "package-name"
+}
+```
+
+* Type: {string}
+
+The `"name"` field defines your package’s name. Node.js doesn't apply any
+restriction on the name field, although the field is ignored if it is not a
+string or an empty string. Other tools (such as
+[npm](https://docs.npmjs.com/files/package.json#name)) may apply other
+restrictions on this field.
+
+The `"name"` field can be used in addition to the [`"exports"`][] field to
+[self-reference][] a package using its name.
+
+### `"type"`
+<!-- YAML
+added: v12.0.0
+changes:
+  - version:
+    - v12.17.0
+    - v13.2.0
+    pr-url: https://github.com/nodejs/node/pull/29866
+    description: Unflag `--experimental-modules`.
+-->
+
+* Type: {string}
+
+The `"type"` field defines the module format `.js` files should be treated
+within a particular `package.json` file’s package scope.
+
+Files ending with `.js` will be loaded as ES modules when the nearest parent
+`package.json` file contains a top-level field `"type"` with a value of
+`"module"`.
+
+The nearest parent `package.json` is defined as the first `package.json` found
+when searching in the current folder, that folder’s parent, and so on up
+until a node_modules folder or the volume root is reached.
+
+```json
+// package.json
+{
+  "type": "module"
+}
+```
+
+```console
+# In same folder as above package.json
+node my-app.js # Runs as ES module
+```
+
+If there is no package.json scope found, or if the nearest parent `package.json`
+lacks a `"type"` field, or contains `"type": "commonjs"`, or any other
+unsupported value, `.js` files are treated as CommonJS.
+
+```js
+// my-app.js, part of the same example as above
+import './startup.js'; // Loaded as ES module because of package.json
+```
+
+Regardless of the value of the `"type"` field, `.mjs` files are always treated
+as ES modules and `.cjs` files are always treated as CommonJS.
+
+### `"exports"`
+<!-- YAML
+added: v12.7.0
+changes:
+  - version:
+    - v12.16.0
+    - v13.2.0
+    pr-url: https://github.com/nodejs/node/pull/29978
+    description: Implement conditional exports.
+  - version:
+    - v12.15.0
+    - v13.7.0
+    pr-url: https://github.com/nodejs/node/pull/31001
+    description: Remove the `--experimental-conditional-exports` option.
+  - version:
+    - v12.16.0
+    - v13.7.0
+    pr-url: https://github.com/nodejs/node/pull/31008
+    description: Implement logical conditional exports ordering.
+-->
+
+```json
+{
+  "exports": "./index.js"
+}
+```
+
+* Type: {Object} | {string} | {string[]}
+
+The `"exports"` field allows defining the [entry points][] of a package when
+imported by name loaded either in a `node_modules` look or via a
+[self-reference][] to its own name. It is supported in Node.js 12+ as an
+alternative to the [`"main"`][] that can support defining [subpath exports][]
+and [conditional exports][].
```

Comment:
Sorry I left out the important encapsulation point here too!

```suggestion
[self-reference][] to its own name. It is supported in Node.js 12+ as an
alternative to the [`"main"`][] that can support defining [subpath exports][]
and [conditional exports][] while encapsulating internal unexported modules.
```

something like that? Open to suggestions.

Reply:
Oh I see, to get rid of the next paragraph. Yeah that's way more concise.